### ISSUE TITLE
dist-upgrade process has failed with centos2alama 1.4.0 and newer

### ISSUE THREAD

**Issue by fbroen** [plesk/centos2alma#332]
I used the newest script from: https://github.com/plesk/centos2alma/issues/330#issuecomment-2293292734

After starting the migration script I get a few minutes later the following error message:
![2024-08-17_error_dist-upgrade_failed](https://github.com/user-attachments/assets/6831e9be-fd0a-4dc8-8c52-033f3dfe2a6d)

The server is a production server only with two websites and no special configuration. I also testet the script with a fresh installed CentOS 7 Server with Plesk and the migration was ok. I don't know why there are so many generated errors, see attached reports:

[leapp-preupgrade.log](https://github.com/user-attachments/files/16645022/leapp-preupgrade.log)
[leapp-report.json](https://github.com/user-attachments/files/16645023/leapp-report.json)
[leapp-report.txt](https://github.com/user-attachments/files/16645024/leapp-report.txt)

Are this bugs of the migration script? The server is running a few years and has allways been updated. The newest plesk version and all CentOS 7.9 updates are installed.

Thank you for your help.
Best regards
Falk

**Comment by SandakovMM** [comment]
The issue likely occurs because the /etc/ssh/sshd_config file lacks the PermitRootLogin directive. Elivate should check this before starting the conversion to ensure you can log in via SSH afterward. The original hint for the precheck is
```
OpenSSH configuration file does not explicitly state the option PermitRootLogin in sshd_config file, which will default in RHEL8 to "prohibit-password".
```
To resolve this, add the following directive to /etc/ssh/sshd_config:
```
PermitRootLogin yes
```
I will also include this pre-check in centos2alma.

**Comment by SandakovMM** [comment]
> But what's with the other problems with "Risk Factor : high"? Do these issues need to be fixed manually, or can the upgrade script fix them automatically?

Usually, they do not cause problems. However, it could be a problem in your case. For example, if you use `python` somewhere. I suggest checking them just to be sure.

**Comment by fbroen** [comment]
Hi @SandakovMM thank you again for the information. So if I only use PHP and MariaDB all listed Risk Factors can be ignored or do not show again if I fix the SSH config? The migration should work only with fixing the ssh config?

**Comment by SandakovMM** [comment]
> Risk Factors can be ignored or do not show again if I fix the SSH config? The migration should work only with fixing the ssh config?

Yes, you right.

**Comment by fbroen** [comment]
Hi @SandakovMM thank you very much for the info again. When do you release the next official script version?

**Comment by SandakovMM** [comment]
> When do you release the next official script version?

I will work on the new version until mid-September, aiming for release on the 16th of September

**Comment by SandakovMM** [comment]
I have included the pre-checker I mentioned in #335. Now, if `PermitRootLogin` is missing in `/etc/ssh/sshd_conf`, the conversion will be prevented by centos2alma itself.
